pkg/proxy/iptables/proxier.go

```diff
@@ -408,14 +410,16 @@ func (proxier *Proxier) Sync() {
 	proxier.mu.Lock()
 	defer proxier.mu.Unlock()
 	proxier.syncProxyRules()
+	proxier.timer.Stop()
+	proxier.timer.Reset(proxier.syncPeriod)
 }
 
 // SyncLoop runs periodic work.  This is expected to run as a goroutine or as the main loop of the app.  It does not return.
 func (proxier *Proxier) SyncLoop() {
-	t := time.NewTicker(proxier.syncPeriod)
-	defer t.Stop()
+	proxier.timer = time.NewTimer(proxier.syncPeriod)
+	defer proxier.timer.Stop()
 	for {
-		<-t.C
+		<-proxier.timer.C
```

Comment:
if someone refactors the resets() embedded *somewhere* in the call path, this timer will hang. Coupling like that should always be encapsulated in a struct so some eager newbie doesn't shoot themselves in the foot. Please add a comment explaining where we reset it.